src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,534 @@
+                        jsonSerializerSettings);
+                    var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                    careBundles.Add(careBundle);
+                }
+            }
+
+            if (tmhPatientData.AllergiesData != null && tmhPatientData.AllergiesData.Any())
+            {
+                var allergiesData =
+                    FetchAllergiesData(dataRequest, tmhPatientData.AllergiesData, patientName).GetAwaiter().GetResult();
+                if (allergiesData.HasValue)
+                {
+                    var serializeObject = JsonConvert.SerializeObject(
+                        allergiesData.ValueOr((ObservationResponse) null),
+                        jsonSerializerSettings);
+                    var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                    careBundles.Add(careBundle);
+                }
+            }
+
+            if (tmhPatientData.AbdomenExaminationsData != null && tmhPatientData.AbdomenExaminationsData.Any())
+            {
+                var abdomenExaminationsData = FetchAbdomenExaminationData(dataRequest,
+                    tmhPatientData.AbdomenExaminationsData, patientName).GetAwaiter().GetResult();
+                if (abdomenExaminationsData.HasValue)
+                {
+                    var serializeObject =
+                        JsonConvert.SerializeObject(abdomenExaminationsData.ValueOr((ObservationResponse) null),
+                            jsonSerializerSettings);
+                    var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                    careBundles.Add(careBundle);
+                }
+            }
+
+            if (tmhPatientData.OralCavityExaminationsData != null && tmhPatientData.OralCavityExaminationsData.Any())
+            {
+                var oralCavityExaminationsData = FetchOralCavityExaminationsData(dataRequest,
+                    tmhPatientData.OralCavityExaminationsData, patientName).GetAwaiter().GetResult();
+                if (oralCavityExaminationsData.HasValue)
+                {
+                    var serializeObject = JsonConvert.SerializeObject(
+                        oralCavityExaminationsData.ValueOr((ObservationResponse) null),
+                        jsonSerializerSettings
+                    );
+                    var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                    careBundles.Add(careBundle);
+                }
+            }
+
+            if (tmhPatientData.SurgeryHistories == null || !tmhPatientData.SurgeryHistories.Any()) return careBundles;
+            {
+                var surgeryHistoriesData =
+                    FetchSurgeryHistoryData(dataRequest, tmhPatientData.SurgeryHistories, patientName).GetAwaiter()
+                        .GetResult();
+                if (!surgeryHistoriesData.HasValue) return careBundles;
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)